### PR TITLE
Added code to identify the Linux distribution

### PR DESCRIPTION
**Pull request by alanjjenkins**
Hey guys,

I have added some code to identify what the Linux distribution in use. I am adding this code as there is a nasty hack in place in one part of ATLauncher that is preventing Linux users from being able to have the Config, Downloads, etc folders in a different folder.

The hack was apparently added to fix ATLauncher crashing on a Linux distribution but I can find no information as to which distribution and after having tested ATLauncher with the hack removed on most of the main distributions I have yet to see it crash. This code will hopefully help us identify what the problematic distribution is and then figure out what about it causes the problem.

Also in this pull request is a small tweak to the build documentation that shows what is needed to be installed on Arch Linux for building ATLauncher.

I will also be submitting a pull request to remove the previously mentioned hack and add the ability to set the location of the Downloads directory and others via environment variables as soon as I finish my testing of the code.

Thanks,

Alan

### CODE REVIEW

**Comment by demon012** [comment]
Also I should add that this has been tested on:

Arch Linux, Debian, Ubuntu, Gentoo and Fedora and it reports the correct operating system and where possible the release (Neither Arch or Gentoo have a release as they are both rolling release distributions).

File: src/main/java/com/atlauncher/App.java [hunk 1]
```diff
@@ -165,7 +169,60 @@ public void run() {
...
+                        if(line.indexOf("PRETTY_NAME") != -1) {
+                            distro = line.substring(12).replace("\"","");
+                        }
+                    }
+                }
+                catch(IOException e) {
+                    LogManager.warn("Unable to read os-release:" + e.getMessage());
+                }
+            }
+            else if(new File("/etc/redhat-release").exists())
```
Comment (lexware) [inline src/main/java/com/atlauncher/App.java]: Can't you just use System.getProperty("os.name");  to get the os

Comment (demon012) [inline src/main/java/com/atlauncher/App.java]: It gets the OS but that will just state Linux, Linux is the operating
system but the point of this code is finding out what Linux Distribution is
in use and if possible the release of the distribution.

**Comment by RyanTheAllmighty** [comment]
Overall being able to identify the specific linux distro is great, but I did make a couple comments :)

**Comment by RyanTheAllmighty** [comment]
Sorry for the delays, but I just cannot accept this PR due to

https://github.com/demon012/ATLauncher/blob/06b6084c78fb5ec7f91fd302b3f29ccb7a3702a9/src/main/java/com/atlauncher/utils/Utils.java#L166

This is going to break some peoples installs and make them yell at me :(

**Comment by demon012** [comment]
We have to remove that. That breaks Linux support not improves it. By
having that line in the code you make it so there is no way of setting the
path to anything other than the directory that the JAR file is in. It
ignores the current working directory and any other set environment
variables.

Please remove it and send anyone who is complaining to me as it is either
their JVM, or PEBCAK that is causing the problem not Linux or most JVMs.

As previously mentioned I tested most of the main distributions with these
changes and they all work fine so.

**Comment by RyanTheAllmighty** [comment]
I'll have to recheck this, as every single time I tested when this all first got up, everything was shoved in the users home directory rather than the folder of the jar as intended.

You've got to understand, when dealing with something which could completely break thousands of end users installs, it's definitely something I'm cautious about.

**Comment by demon012** [comment]
I understand that but the current behaviour is really a bug as it makes the
files appear wherever the jar file is which upsets a lot of people in Linux
world as they have standards for where particular types of files should be
and with what permissions.

I have for a long time just had a rather open set of permissions set on the
Arch Linux package but it isn't right and there are a few people moaning
about it.

May I ask what distro you were testing on that put the files in your home
folder?

Also keep in mind that the environment variables have to override the
setting for checking if there are files in the same directory (should be
safe as if you are using the documentation and know how to use environment
variables you should know how to use the launcher).

I also need to work on some documentation for these changes so people are
aware of the environment variables and how to use them.

**Pull request closed without merging**